pulsar-functions/localrun/src/main/java/org/apache/pulsar/functions/LocalRunner.java

```diff
@@ -196,18 +200,37 @@ public LocalRunner(FunctionConfig functionConfig, SourceConfig sourceConfig, Sin
         this.runtimeEnv = runtimeEnv;
         this.secretsProviderClassName = secretsProviderClassName;
         this.secretsProviderConfig = secretsProviderConfig;
-        this.narExtractionDirectory = narExtractionDirectory != null ? narExtractionDirectory
-                : NarClassLoader.DEFAULT_NAR_EXTRACTION_DIR;
+        if (narExtractionDirectory != null) {
```

Comment:
We are changing the existing behavior here. If the narExtractionDirectory is not specified we should default to NarClassLoader.DEFAULT_NAR_EXTRACTION_DIR

Reply:
The change in behavior is intentional for LocalRunner. The intent of the changes are that unless narExtractionDirectory is specified, a temporary directory will be created which is deleted when `LocalRunner.close()` is called. Do you think that this type of behavior is unacceptable? I hope it would be ok to do this change since it reduces the chance that temporary files get left in the main system temp directory.

Reply:
I think it is fine